20_classes_lab/03.py

```diff
@@ -1,13 +1,46 @@
-"""
-Following code assumes a class named Widget that depends on
-other widgets
-Widget ctor takes a name, and function 
-add_dependency(depends)
-marks widget as depends on all widgets in "depends" list
-
-When building a widget, you need to first build all the widgets it
-depends on.
-
-Implement widget so the following code works
-"""
-
+"""
+Classes and objects no 2
+"""
+import sys
+import os
+
+class Widget(object):
+
+    all_objects = []

```

Comment:
Keeping track of all built object is a waste. If an object should leave scope and needs to be deleted, memory won't be freed up because it has a copy in all_objects.

Also iterating an array of all the widgets ever built just to find out if a given widget was already built is a waste of CPU time.

Please try to find a better solution without the class variable